src/components/AttachmentModal.js

```diff
@@ -102,14 +106,21 @@ function AttachmentModal(props) {
             : undefined,
     );
 
+    const onCarouselAttachmentChange = props.onCarouselAttachmentChange;
+
     /**
      * Keeps the attachment source in sync with the attachment displayed currently in the carousel.
      * @param {{ source: String, isAuthTokenRequired: Boolean, file: { name: string } }} attachment
      */
-    const onNavigate = useCallback((attachment) => {
-        setSource(attachment.source);
-        setFile(attachment.file);
-    }, []);
+    const onNavigate = useCallback(
+        (attachment) => {
+            setSource(attachment.source);
+            setFile(attachment.file);
+            setIsAuthTokenRequired(attachment.isAuthTokenRequired);
+            onCarouselAttachmentChange(attachment);
+        },
+        [onCarouselAttachmentChange],
+    );
```

Comment:
Perhaps it's necessary, but it's a bit confusing to me that we're both subscribing to changes from, and invoking `onCarouselAttachmentChange`. Is this correct @kidroca?

Reply:
Hi @Julesssss, thanks for your review. I see where the confusion is coming from. In this case, we're using the `useCallback` hook. The `onCarouselAttachmentChange` is a dependency of this hook, meaning that the function will only change if `onCarouselAttachmentChange` changes. This is not a subscription in the traditional sense, but rather a way to ensure that our function always has the most up-to-date version of `onCarouselAttachmentChange`. I hope this clears up the confusion.

Reply:
Thanks for explaining, after reading it a few times I think that makes sense now 😅